### PR TITLE
allow for type attribute

### PR DESCRIPTION
**Pull request by lanej**

### CODE REVIEW

File: test/serializers/attribute_test.rb [hunk 1]
```diff
@@ -33,6 +33,21 @@ def test_multiple_calls_with_the_same_attribute
 
         assert_equal([:title], serializer_class._attributes)
       end
+
+      def test_type_attribute_override
+        attribute_serializer = Class.new(ActiveModel::Serializer) do
+          attribute :id, key: :type
```
Comment (bf4) [inline test/serializers/attribute_test.rb]: nitpick: it's not an override anymore if the internal name is now 'json_api_type', right?

Comment (lanej) [inline test/serializers/attribute_test.rb]: very true.

Comment (bf4) [inline test/serializers/attribute_test.rb]: 😇

**Comment by joaomdmoura** [comment]
@lanej Could you please rebase it? :grin: 
As I said, great work here, I was looking forward to it! :+1:

**Comment by lanej** [comment]
@joaomdmoura rebased.

**Comment by joaomdmoura** [comment]
Great, I'm merging it.

**Pull request merged**